app/controllers/users_controller.rb

```diff
@@ -13,6 +13,12 @@ def create
     if @user.save
       flash[:alert] = "You are now registered and logged in as #{@user.email}."
       redirect_to profile_path
+    elsif @user.errors.keys.include?(:email) &&
```

Comment:
What if we changed line 16-18 to: `elsif User.find_by(email: @user.email)`

Reply:
Spoke with Blake, the intention with this was to use the errors to avoid querying the database again.